### DISCUSSION TITLE
Welcome to g3log Discussions!

### DISCUSSION THREAD

**Discussion by KjellKod** [KjellKod/g3log#509] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about. I.e this can be a way to discuss a wanted feature for feedback rather than opening up a ticket. 
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with a short introduction of yourself and tell us about your thoughts.

**Comment by VishalGujela** [comment]
Hello,
I was just curious to know that , can we use g3log to create separate log files for each module or services that i'll be running.
And also i'll be customising my own severity levels.
So , can we perform such tasks using this library.
Thanks

**Reply by KjellKod** [reply]
You can create your own logging levels. Please see the API documentation for details as well as the level unit tests

G3log is made to be easy to extend with your own logging requirements. Adding a custom made sink is extremely easy. For logging requirements that is specific to your company, you can either copy the existing logged sink in g3log or maybe one of the filtering sinks in the g3sinks would be a better starting point? 

- that said. G3log does not have a sink right now that does logging per file, module or per read. 
- I'd suggest using the advice I gave you above and use one custom logging sink that writes to multiple files according to your own specifications. File details is already available to a sink through the received LogMessage.
